spec/services/applications/revert_to_pending_spec.rb

```diff
@@ -0,0 +1,131 @@
+require "rails_helper"
+
+RSpec.describe Applications::RevertToPending, type: :model do
+  subject(:instance) { described_class.new(application) }
+
+  let :application do
+    create(:application, :accepted) do |application|
+      create(:application_state, application:)
+    end
+  end
+
+  describe ".call" do
+    subject(:call_service) { described_class.call(application) && application.reload }
+
+    it "updates lead provider approval status" do
+      expect { call_service }
+        .to change { application.reload.lead_provider_approval_status }
+                   .from("accepted")
+                   .to("pending")
+    end
+
+    it "empties the funded_place attribute" do
+      expect { call_service }
+        .to change { application.reload.funded_place }
+                   .from(false)
+                   .to(nil)
+    end
+
+    it "removes Application states" do
+      expect { call_service }
+        .to change { application.application_states.count }
+                   .from(1)
+                   .to(0)
+    end
+
+    context "when already pending" do
+      let(:application) { create(:application, :pending) }
+
+      it "succeeds but does not change the approval status" do
```

Comment:
Worth also testing it doesn't change application states/declarations/funded place in this scenario?